homeassistant/components/smart_meter_b_route/config_flow.py

```diff
@@ -0,0 +1,75 @@
+"""Config flow for Smart Meter B Route integration."""
+
+import logging
+from typing import Any
+
+from momonga import Momonga, MomongaSkJoinFailure, MomongaSkScanFailure
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_DEVICE, CONF_ID, CONF_PASSWORD
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN, ENTRY_TITLE
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_DEVICE): str,
+        vol.Required(CONF_ID): str,
+        vol.Required(CONF_PASSWORD): str,
+    }
+)
+
+
+def validate_input(device: str, id: str, password: str) -> None:
+    """Validate the user input allows us to connect."""
+    try:
+        with Momonga(dev=device, rbid=id, pwd=password):
+            pass
+    except MomongaSkScanFailure as error:
+        raise CannotConnect from error
+    except MomongaSkJoinFailure as error:
+        raise InvalidAuth from error
```

Comment:
Honestly, remove the try catch here, and just catch the Momonga exceptions in the form instead of the CannotConnect and InvalidAuth